mpt/src/mpt.rs

```diff
@@ -0,0 +1,1668 @@
+                                |branch_acc: &mut F,
+                                 branch_mult: &mut F,
+                                 rlp_start: usize,
+                                 start: usize| {
+                                    if row[rlp_start + 1] == 0 {
+                                        *branch_acc += c128 * *branch_mult;
+                                        *branch_mult *= self.acc_r;
+                                    } else {
+                                        *branch_acc += c160 * *branch_mult;
+                                        *branch_mult *= self.acc_r;
+                                        for i in 0..HASH_WIDTH {
+                                            *branch_acc +=
+                                                F::from(row[start + i] as u64) * *branch_mult;
+                                            *branch_mult *= self.acc_r;
+                                        }
+                                    }
+                                };
+
+                            // TODO: add branch ValueNode info
+
+                            compute_branch_acc_and_mult(
+                                &mut pv.acc_s,
+                                &mut pv.acc_mult_s,
+                                S_RLP_START,
+                                S_START,
+                            );
+                            compute_branch_acc_and_mult(
+                                &mut pv.acc_c,
+                                &mut pv.acc_mult_c,
+                                C_RLP_START,
+                                C_START,
+                            );
+                            self.assign_acc(
+                                &mut region,
+                                pv.acc_s,
+                                pv.acc_mult_s,
+                                pv.acc_c,
+                                pv.acc_mult_c,
+                                offset,
+                            )?;
+
+                            offset += 1;
+                            pv.node_index += 1;
+                        } else if row[row.len() - 1] == 2
+                            || row[row.len() - 1] == 3
+                            || row[row.len() - 1] == 4
+                            || row[row.len() - 1] == 6
+                            || row[row.len() - 1] == 7
+                            || row[row.len() - 1] == 8
+                            || row[row.len() - 1] == 9
+                            || row[row.len() - 1] == 10
+                            || row[row.len() - 1] == 11
+                            || row[row.len() - 1] == 13
+                            || row[row.len() - 1] == 14
+                            || row[row.len() - 1] == 15
+                            || row[row.len() - 1] == 16
+                            || row[row.len() - 1] == 17
+                            || row[row.len() - 1] == 18
+                        {
```

Comment:
We can simplify this expression
```suggestion
                        } else if (2..=18).contains(&row[row.len() - 1]) {
```

Reply:
Yes, much needed. At some point there were more types not used (now all are used except 5 which marks that the row only needs to be hashed and filled into keccak_table).